### PR TITLE
Enable WASM AOT microbenchmark run

### PR DESCRIPTION
**Pull request by Lxiamail**

### CODE REVIEW

**Review by naricc — APPROVED** [review]

File: eng/testing/performance/performance-setup.sh [hunk 1]
```diff
@@ -264,7 +269,16 @@ if [[ "$wasm_runtime_loc" != "" ]]; then
...
     wasm_dotnet_path=$payload_directory/dotnet-wasm
     mv $wasm_runtime_loc $wasm_dotnet_path
-    extra_benchmark_dotnet_arguments="$extra_benchmark_dotnet_arguments --wasmMainJS \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm/runtime-test.js --wasmEngine /home/helixbot/.jsvu/$javascript_engine --customRuntimePack \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm"
+    if [[ "$wasmaot" == "true" ]]; then
+        # wasm aot needs some source code from dotnet\runtime repo
+        rsync -aq --progress $source_directory/* $wasm_dotnet_path --exclude payload --exclude docs --exclude src/coreclr --exclude src/tests --exclude artifacts/obj
+        # copy wasm build drop to the location that aot build expects
+        rsync -a --progress $wasm_dotnet_path/artifacts/BrowserWasm/artifacts/* $wasm_dotnet_path/artifacts
+        rm -r $wasm_dotnet_path/artifacts/BrowserWasm/artifacts
+        extra_benchmark_dotnet_arguments="$extra_benchmark_dotnet_arguments --wasmEngine /home/helixbot/.jsvu/$javascript_engine --aotcompilermode wasm --runtimeSrcDir \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm --buildTimeout 3600 --keepfiles"
```
Comment (DrewScoggins) [inline eng/testing/performance/performance-setup.sh:278]: Make sure that --keep-files is removed

File: eng/pipelines/coreclr/templates/perf-job.yml [hunk 2]
```diff
@@ -54,9 +54,9 @@ jobs:
       - ${{ format('libraries_build_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.liveLibrariesBuildConfig) }}
     - ${{ if and(eq(parameters.runtimeType, 'mono'), ne(parameters.codeGenType, 'AOT')) }}:
       - ${{ format('mono_{0}_product_build_{1}{2}_{3}_{4}', parameters.runtimeVariant, parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
-    - ${{ if eq(parameters.runtimeType, 'wasm') }}:
-      - ${{ format('build_{0}{1}_{2}_{3}_{4}', 'Browser', '', 'wasm', parameters.buildConfig, parameters.runtimeType) }}
-    - ${{ if eq(parameters.codeGenType, 'AOT')}}:
+    - ${{ if eq(parameters.runtimeType, 'wasm')}}:
+      - ${{ format('build_{0}{1}_{2}_{3}_{4}', 'Browser', '', 'wasm', parameters.buildConfig, parameters.runtimeType, parameters.codeGenType) }}
```
Comment (DrewScoggins) [inline eng/pipelines/coreclr/templates/perf-job.yml:58]: We never reference the codeGenType parameter, it is the 6th one, so we would need to see {5} in the string, but don't.

File: eng/testing/performance/performance-setup.sh [hunk 3]
```diff
@@ -249,12 +259,18 @@ common_setup_arguments="--channel main --queue $queue --build-number $build_numb
...
 
 if [[ "$run_from_perf_repo" = true ]]; then
+    echo "Running from perf repo"
     payload_directory=
     workitem_directory=$source_directory
     performance_directory=$workitem_directory
     setup_arguments="--perf-hash $commit_sha $common_setup_arguments"
 else
-    git clone --branch main --depth 1 --quiet https://github.com/dotnet/performance $performance_directory
+    echo "Not running from perf repo"
```
Comment (Lxiamail) [inline eng/testing/performance/performance-setup.sh:268]: line 263-269 should be revert back to the original code to clone main branch once PR https://github.com/dotnet/performance/pull/1902 is merged

File: eng/testing/performance/performance-setup.sh [hunk 4]
```diff
@@ -228,7 +234,11 @@ if [[ "$mono_dotnet" != "" ]] && [[ "$monointerpreter" == "false" ]]; then
 fi
 
 if [[ "$wasm_runtime_loc" != "" ]]; then
-    configurations="CompilationMode=wasm RunKind=$kind"
+    if [[ "$wasmaot" == "true" ]]; then
+        configurations="CompilationMode=wasm AOT=true RunKind=$kind"
```
Comment (DrewScoggins) [inline eng/testing/performance/performance-setup.sh:238]: AOT -> MonoAOT to match the current configs we have.

File: eng/testing/performance/performance-setup.sh [hunk 5]
```diff
@@ -254,8 +264,10 @@ if [[ "$run_from_perf_repo" = true ]]; then
     performance_directory=$workitem_directory
     setup_arguments="--perf-hash $commit_sha $common_setup_arguments"
 else
-    git clone --branch main --depth 1 --quiet https://github.com/dotnet/performance $performance_directory
-    
+    git clone --branch alicial/wasmaotmicro --depth 1 https://github.com/dotnet/performance.git $performance_directory
```
Comment (DrewScoggins) [inline eng/testing/performance/performance-setup.sh:267]: This needs to switch back to cloning from the performance repo.

File: eng/testing/performance/performance-setup.sh [hunk 6]
```diff
@@ -264,7 +276,21 @@ if [[ "$wasm_runtime_loc" != "" ]]; then
     using_wasm=true
     wasm_dotnet_path=$payload_directory/dotnet-wasm
     mv $wasm_runtime_loc $wasm_dotnet_path
-    extra_benchmark_dotnet_arguments="$extra_benchmark_dotnet_arguments --wasmMainJS \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm/runtime-test.js --wasmEngine /home/helixbot/.jsvu/$javascript_engine --customRuntimePack \$HELIX_CORRELATION_PAYLOAD/dotnet-wasm"
+    # install emsdk, $source_directory/src/mono/wasm/ has the nuget.config with require feed. EMSDK may be available in the payload in a different directory, should visit this install to avoid deplicated payload.
+    pushd $source_directory/src/mono/wasm/
+    make provision-wasm
+    EMSDK_PATH = $source_directory/src/mono/wasm/emsdk
```
Comment (DrewScoggins) [inline eng/testing/performance/performance-setup.sh:282]: This variable gets set, but it does not ever seem to get used anywhere. What is it for?

**Review by DrewScoggins — APPROVED** [review]
LGTM, apart from the few small things I mentioned in comments. ~~Also we need to remove the performance submodule~~

File: eng/testing/performance/performance-setup.sh (hunk 6 above)
Comment (Lxiamail) [reply eng/testing/performance/performance-setup.sh:282]: Yes, EMSDK_PATH is not used in our script file. It is set here is per document guidance at https://github.com/dotnet/runtime/blob/d953229d5429d2ffde833740dd481aab864d3e0c/src/mono/wasm/README.md

File: eng/testing/performance/performance-setup.sh (hunk 4 above)
Comment (Lxiamail) [reply eng/testing/performance/performance-setup.sh:238]: Is MonoAOT different from wasm aot?

**Pull request merged**